surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProvider.java

```diff
@@ -173,15 +173,17 @@ private void invokeAllTests(TestsToRun testsToRun, RunListenerAdapter adapter) {
 
     private void execute(TestsToRun testsToRun, RunListenerAdapter adapter) {
         if (testsToRun.allowEagerReading()) {
-            List<DiscoverySelector> selectors = new ArrayList<>();
-            testsToRun.iterator().forEachRemaining(c -> selectors.add(selectClass(c.getName())));
+            testsToRun.iterator().forEachRemaining(c -> {
+                List<DiscoverySelector> selectors = new ArrayList<>();
+                selectors.add(selectClass(c.getName()));
 
-            LauncherDiscoveryRequestBuilder builder = request()
-                    .filters(filters)
-                    .configurationParameters(configurationParameters)
-                    .selectors(selectors);
+                LauncherDiscoveryRequestBuilder builder = request()
+                        .filters(filters)
+                        .configurationParameters(configurationParameters)
+                        .selectors(selectors);
 
-            launcher.execute(builder.build(), adapter);
+                launcher.execute(builder.build(), adapter);
+            });
```

Comment:
Please call me stupid, but where is the difference to the `else` block now? This just outfactored the selectors to a list and is created fresh with each item in the iterator. Identical to the `else` block, no?

Reply:
Oh you are right, I completely missed that, will clean this up.